frontend/src/explorer/Explorer.tsx

```diff
@@ -21,9 +21,10 @@ const propTypes = {
   onExpand: PropTypes.func.isRequired,
   selection: PropTypes.object,
   setSelection: PropTypes.func.isRequired,
+  readOnly: PropTypes.bool,
```

Comment:
Remove proptypes and switch to a proper TypeScript type for the component instead since you are modifying the props. Thanks :)

Reply:
Done